src/app/pulse/charts/chart-area.component.html

```diff
@@ -1,13 +1,14 @@
-<div *ngIf="!charts[0]?.datasets?.length || error" class="no-charts">
+<div *ngIf="!charts[0]?.datasets?.length || error || !charts[0]?.datasets[0]?.data.length" class="no-charts">
```

Comment:
well, the logic inside ngIfs became complicated really quick). Maybe just extract them from the template to getters or methods?